### PR TITLE
Support reloading CA files

### PR DESCRIPTION
**Pull request by deads2k**
I understand how the configForClient function works now.  I've updated the code to use it and provided a way to have the authenticators reload.

I still need to wire the aggregator for reloading, but I think I can force that based on a value change with another patch.

/assign @sttts @mfojtik

### CODE REVIEW

File: pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch_authenticator.go [hunk 1]
```diff
@@ -1,10 +1,11 @@
 package openshiftkubeapiserver
 
 import (
-	"crypto/x509"
 	"fmt"
 	"time"
 
+	"k8s.io/apiserver/pkg/server/certs"
+
```
Comment (mfojtik) [inline pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch_authenticator.go]: nit: remove

**Comment by mfojtik** [comment]
LGTM

@sttts feel free to tag

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/certs/serving.go [hunk 2]
```diff
@@ -26,17 +27,25 @@ type CertKeyFileReference struct {
...
 }
 
+type CABundleFileReferences struct {
+	CABundles []string
+}
+
 // DynamicLoader dynamically loads certificates and provides a golang tls compatible dynamic GetCertificate func.
-type DynamicLoader struct {
+type DynamicServingLoader struct {
+	BaseTLSConfig tls.Config
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/certs/serving.go]: this one could have a go doc. What is it? What is used from it? What is overridden?

File: vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go [hunk 3]
```diff
@@ -17,8 +17,11 @@ limitations under the License.
 package authenticator
 
 import (
+	"fmt"
 	"time"
 
+	"k8s.io/apiserver/pkg/server/certs"
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/pkg/kubeapiserver/authenticator/config.go]: nit: order

File: vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/config.go [hunk 4]
```diff
@@ -234,6 +234,8 @@ type AuthenticationInfo struct {
 	// If this is true, a basic auth challenge is returned on authentication failure
 	// TODO(roberthbailey): Remove once the server no longer supports http basic auth.
 	SupportsBasicAuth bool
+
+	DynamicReloadFns map[string]PostStartHookFunc
```
Comment (sttts) [inline vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/config.go]: what are these?

**Comment by sttts** [comment]
LGTM. Would like to see two godocs. Feel free to self-tag.

**Comment by deads2k** [comment]
> LGTM. Would like to see two godocs. Feel free to self-tag.

Will add after

**Comment by deads2k** [comment]
Green. Will add followups.  Merging to unstick the openshiftapiserveroperator.

**Pull request merged**